BinaryTree.java

```diff
@@ -0,0 +1,168 @@
+package com.katermar;
+
+/**
+ * Created by USER on 5/23/2017.
+ */
+public class BinaryTree {
+    TreeNode root;
+
+    /*Inner class for Binary Tree node
+    * @field value stores integer value
+    * @field left, right store left and right subtrees*/
+    public static class TreeNode {
+        private int value;
+        private TreeNode left, right, nextRight;
+
+        TreeNode(int item) {
+            value = item;
+            left = right = nextRight = null;
+        }
+
+        public int getValue() {
+            return value;
+        }
+
+        public TreeNode getLeft() {
+            return left;
+        }
+
+        @Override
+        public String toString() {
+            return "TreeNode{" +
+                    "value=" + value +
+                    '}';
+        }
+
+        public TreeNode getRight() {
+            return right;
+        }
+
+        public TreeNode getNextRight() {
+            return nextRight;
+        }
+    }
+
+    /* Function to print all ancestors of purposed value
+    * @param node - root node
+    * @param value - value of the node on which we want to print ancestors
+    * @return true, if value exists in this tree, otherwise - false*/
+    boolean printAllAncestors(TreeNode node, int value)
+    {
+         //if tree is empty returns false
+        if (node == null)
+            return false;
+
+        //if value exist in a tree
+        if (node.getValue() == value)
+            return true;
+
+        //if value is present in left/right subtree
+        // then we'll print it
+        if (printAllAncestors(node.left, value) || printAllAncestors(node.right, value))
+        {
+            System.out.print(node.value + " ");
+            return true;
+        }
+
+        //else return false
+        return false;
+    }
+
+    /*Traverse tree in preorder
+    * @param root - tree root*/
+    public void preorderTraversal (TreeNode root){ //метод для обхода в ширину
+        if (root != null){
+            System.out.print(root.getValue() + " ");
+            preorderTraversal (root.left);
+            preorderTraversal (root.right);
+        }
+    }
+
+    /*Populates tree with the array in order
+    * @param array - array which will fill the tree*/
+    public boolean populateTree(int[] array) {
+        for (int i = 0; i < array.length; i++) {
+            this.insert(array[i]);
+        }
+
+        return true;
+    }
+
+    /*@param value - value to insert*/
+    public void insert(int value) {
+        root = insert(root, value);
+    }
+
+    /*Inserts value to the tree
+    * @param node - root
+    * @param value - value to insert
+    * @return root*/
+    private static TreeNode insert(TreeNode node, int value) {

```

Comment:
Yeah I know that you didn't solve it assuming it will be a binary _search_ tree, but someone asked this question on Slack and this is the reply he got:
![image](https://cloud.githubusercontent.com/assets/17977547/26449059/eba276b4-4158-11e7-8265-d7d5181fcf90.png)
Just wanted to point this out :)